src/chain/chaininterface.rs

```diff
@@ -75,7 +79,7 @@ pub trait ChainListener: Sync + Send {
 	///
 	/// This also means those counting confirmations using block_connected callbacks should watch
 	/// for duplicate headers and not count them towards confirmations!
-	fn block_connected(&self, header: &BlockHeader, height: u32, txn_matched: &[&Transaction], indexes_of_txn_matched: &[u32]);
+	fn block_connected(&self, header: &BlockHeader, height: u32, txn_matched: &[Transaction], indexes_of_txn_matched: &[u32]);
```

Comment:
Hmm, why force a Transaction itself? Its cheap to create &[&Transaction] from &[Transaction], but not the other way (and, eg, you may imagine something filtering blocks by just walking the tx set and adding references to the ones that match to a Vec, instead of making a new Vec and copying the full Transactions that match.

It should be relatively straightforward to just make get_watched_txs into get_watched_txs(&self, block: &'a Block) -> (Vec<&'a Transaction>, Vec<u32>) or so.